src/client/Microsoft.Identity.Client/ApiConfig/BrokerOptions.cs

```diff
@@ -0,0 +1,116 @@
+        public BrokerOptions(OperatingSystems operatingSystems)
+        {
+            if (operatingSystems == OperatingSystems.None)
+            {
+                throw new ArgumentException($"Operating system must be specified.");
+            }
+        }
+
+        // The default constructor is private. So developer is forced to set the OS choice(s)
+        private BrokerOptions()
+        {
+
+        }
+
+        /// <summary>
+        /// Creates default options that can be modified later except the choice of OS
+        /// </summary>
+        /// <param name="osChoice">Choice of OS platforms</param>
+        /// <param name="listWorkAndSchoolAccts">List wokr and school accounts</param>
+        /// <returns></returns>
+        public static BrokerOptions CreateDefault()
+        {
+            return new BrokerOptions(osChoice)
+            {
+                 OSChoice = OperatingSystems.Windows,
+                 ListOperatingSystemAccounts = true,
+            };
+        }
+
+        /// <summary>
+        /// Creates BrokerOptions from WindowsBrokerOptions
+        /// </summary>
+        /// <param name="winOptions"></param>
+        /// <param name="osChoice"></param>
+        /// <returns></returns>
+        internal static BrokerOptions CreateFromWindowsOptions(WindowsBrokerOptions winOptions, OperatingSystems osChoice = OperatingSystems.Windows)
+        {
+            BrokerOptions ret = new BrokerOptions(osChoice);
+            ret.Title = winOptions.HeaderText;
+            ret.MsaPassthrough = winOptions.MsaPassthrough;
+            ret.ListOperatingSystemAccounts = winOptions.ListWindowsWorkAndSchoolAccounts;
+
+            return ret;
+        }
+
+        /// <summary>
+        /// This is a required property to determine the supported OS
+        /// </summary>
+        public OperatingSystems OSChoices { get; private set; }
+
+        /// <summary>
+        /// Title of the broker window
+        /// </summary>
+        public string Title { get; set; }
+
+        /// <summary>
+        /// A legacy option available only to Microsoft applications. Should be avoided where possible.
+        /// Support is experimental.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)] // 1p feature only, hide it from public API.
+        public bool MsaPassthrough { get; set; } = false;
+
+        /// <summary>
+        /// Currently only supported on Windows
+        /// Allow the Windows broker to list Work and School accounts as part of the <see cref="ClientApplicationBase.GetAccountsAsync()"/>
+        /// </summary>
+        /// <remarks>On UWP, accounts are not listed due to privacy concerns</remarks>/// 
+        public bool ListOperatingSystemAccounts { get; set; }
+
+        /// <summary>
+        /// This is to validate the options
+        /// </summary>
+        internal void Validate()
+        { 
+            if(OSChoices == OperatingSystems.None)
```

Comment:
Why would we fail on this? It simply means broker is not enabled.

Reply:
Makes sense. Not needed as this is assigned in the constructor and set is private. Removed